### ISSUE TITLE
Tf.dataset shuffle buffer cannot release

### ISSUE THREAD

**Issue by ziyigogogo** [tensorflow/tensorflow#28759]
**System information**
- TensorFlow version (you are using): 2.0 alpha gpu

**Describe the feature and the current behavior/state.**
- When using shuffle op of tf.dataset, the buffer cannot release when the dataset instance is deleted. 
For example, i train my model with 2 datasets in sequence.
dataset1.shuffle(buffer_size = 1000)
model.fit(dataset1)   memory consumption: 50%
del dataset1
dataset2.shuffle(buffer_size = 1000) memory consumption: 80%
model.fit(dataset2)

This is not a matter when shuffle filenames, but serious for img dataset comes form a tfrecord file.
For instance in my case, 1000 imgs(600 X 600 X 3, FP32) in the buffer consumed 30% resource of machine which is really a big matter for automatic train process

**Comment by gadagashwini** [comment]
@ziyigogogo In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by wesboyt** [comment]
I'm able to reproduce whilst running the transformer example verbatim: https://www.tensorflow.org/alpha/tutorials/text/transformer
Nightly is frying my laptop so maybe its already fixed and I just cant tell?
